specs/anchored-region/anchored-region.md

```diff
@@ -0,0 +1,254 @@
+
+For a more in-depth understanding of how this component works under the covers please refer to the [intersection observer api](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). 
+
+### Use Cases
+
+It is envisioned that this component would be used as a building block for other components in this library (select, flyout, tooltip, etc) as well as being available for standalone use in responsive layouts.
+
+### Features
+
+- **Relative positioning:** Users can use to it position an element relative to another another element directly, like enabling a menu to open above or below a trigger button.
+
+- **Responsive positioning:** Users can use it to position an element relative to another element based on available space, for example a menu could open upwards if the trigger button is near the bottom of the page, and downwards if it is nearer the top.
+
+- **Responsive scaling:** Users can use it to create a layout region that dynamically sizes depending on space between the anchor and the viewport elements.
+
+Note that the "always in view" functionality that existed in Viewport Positioner has been removed as this can be replicated through other means (such as a scaling region with grids, for example).
+
+### Risks and Challenges
+
+This component depends on getting accurate positioning data from the dom outside of the component and it is possible for authors to break the relationship in advanced scenarios.
+
+For example can likely only guarantee positioning if there are no other scrolling containers between the viewport and the anchor/anchored-region.  Authors should avoid nested scrolling containers.
+
+Also, some css settings can interfere with the data returned by intersection observer and break the positioning functionality.  For example an element with position="fixed" in the dom hierarchy would be a no-no:
+```
+<div id="viewport">
+    <div style="{{position:fixed}}">
+        <button id="anchor">
+            Button is an anchor
+        </button>
+        <anchored-region
+            viewportid="viewport"
+            anchorid="anchor"
+            verticalPositioningMode="locktodefault"
+            verticalDefaultPosition="top"
+        >
+            This won't work because of the fixed element
+        </anchored-region>
+    <div>
+</div>
+```
+
+### Prior Art/Examples
+- [FAST-DNA React Viewport positioner component](https://github.com/microsoft/fast-dna/tree/master/packages/fast-components-react-base/src/viewport-positioner)
+
+---
+
+## Design
+
+(note: examples are for the vertical axis, equivalent is true for horizontal)
+
+A region that always renders above the anchor element.
+```
+<div id="viewport">
+    ...stuff...
+    <button id="anchor">
+        Button is an anchor
+    </button>
+    <anchored-region
+        anchorid="anchor"
```

Comment:
many attributes expect an ID value without explicitly having "id" in the name - `for="id"`, `aria-labelledby="id"`, `aria-describedby="id"`, etc. Should we just use "anchor" instead of "anchorid"?

Reply:
Sure.